### ISSUE TITLE
Support for PDO.SqlSrv specific attributes

### ISSUE THREAD

**Issue by Indigo744** [peachpiecompiler/peachpie#825]
#813 added the missing SqlSrv constants.

This issue track their support:

- [ ] SQLSRV_ATTR_ENCODING
    - [ ] SQLSRV_ENCODING_DEFAULT
    - [ ] SQLSRV_ENCODING_BINARY
    - [ ] SQLSRV_ENCODING_SYSTEM
    - [ ] SQLSRV_ENCODING_UTF8
- [ ] SQLSRV_ATTR_QUERY_TIMEOUT
- [ ] SQLSRV_ATTR_DIRECT_QUERY
- [ ] SQLSRV_ATTR_CURSOR_SCROLL_TYPE
- [ ] SQLSRV_ATTR_FETCHES_NUMERIC_TYPE
- [ ] SQLSRV_PARAM_OUT_DEFAULT_SIZE

**Comment by Indigo744** [comment]
@jakubmisek I will need some pointers here.

First, maybe we should override `PDOSqlServerDriver.TrySetAttribute()` and throw a `NotImplementedException` so it's clear what is missing?

Next, I would need to support `SQLSRV_ATTR_ENCODING` with `SQLSRV_ENCODING_SYSTEM`, but I have no idea where to start. I cannot find a relevant parameter in the SQL Connection string, no in `DbCommand` or anything...

**Comment by jakubmisek** [comment]
Yes, overring `TrySetAttribute` is the first step, so you can handle the driver-specific attributes.

I think; the value `SQLSRV_ATTR_ENCODING` of has to be stored within the PDO instance and according to it, byte[] eventually decoded to Unicode string. Seems it is not supposed to be passed to DbConnection at all

**Comment by Indigo744** [comment]
> Edit: since we are running on top of .NET database abstraction, this might be already handled and you might ignore it?

The thing is, our code uses `SQLSRV_ENCODING_SYSTEM` which is not UTF8, sadly. So there are some issue with the encoding... And I have no clue how to resolve them!

**Comment by jakubmisek** [comment]
the question is; are the strings coming from SqlSrv API correctly?

**Comment by Indigo744** [comment]
> 
> 
> the question is; are the strings coming from SqlSrv API correctly?

Doesn't seems like it. If I put a breakpoint just after the `fetchAll()`, I can already see encoding artefacts (black question mark) where accent and such should be...
